File: 0_3.leetcode/java/LEET455.java
import java.util.Arrays;

public class LEET455 {
    public static void main(String[] args) {
        System.out.println(findContentChildren(new int[] {1, 2, 3}, new int[] {1, 1}));
    }
    public static int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);

        int result = 0;
        for (int i : s) {
            int index = upperBound(g, i);

            if (index > result) {
                result++;
            } 
        }
        return result;
    }
    public static int upperBound(int[] arr, int target) {
        int left = 0, right = arr.length;

        while (left < right) {
            int mid = left + (right - left) / 2;

            if (arr[mid] <= target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return left;
    }
}

File: 0_3.leetcode/python/LEET455.py
from typing import *
import bisect

class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort(reverse=True)
        s.sort(reverse=True)
        
        i = j = 0
        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                j += 1
            i += 1
        
        return j

    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        
        result = 0 # 쿠키를 부여 받은 아이 수 

        for i in s:
            index = bisect.bisect_right(g, i)

            if index > result: # i 이하의 개수가 준 부여 받은 아이 수 보다 많을 때
                result += 1

        return result